data-pipeline/src/trace_exporter.rs

```diff
@@ -217,16 +217,18 @@ impl TraceExporterBuilder {
             )?,
             api_key: None,
         };
+
         let runtime = tokio::runtime::Builder::new_current_thread()
             .enable_all()
             .build()?;
+
         Ok(TraceExporter {
             endpoint,
             tags: TracerTags {
-                tracer_version: self.tracer_version.clone().unwrap(),
-                language_version: self.language_version.clone().unwrap(),
-                language_interpreter: self.interpreter.clone().unwrap(),
-                language: self.language.clone().unwrap(),
+                tracer_version: std::mem::take(&mut self.tracer_version),
```

Comment:
I haven't run benchmarks, but using `mem::take` is slightly more performant versus `to_owned()` because `to_owned()` does another memory allocation. `mem::take` just swaps the existing value with the default. The downside to this is that the builder doesn't retain values after `build()` is called and thus can cause unexpected results if it is reused. I don't think I've ever reused a builder after I finished building, so I don't think it should be a problem? But, I'm happy to change this to `to_owned()` if people disagree.

Reply:
If the `build` function is _destructive_ in that way, then it should take ownership of the builder with `fn build(self) ...`, to avoid misuse.